segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLogFactory.java

```diff
@@ -26,18 +35,31 @@
 import org.apache.bookkeeper.net.CommonConfigurationKeys;
 import org.apache.curator.framework.CuratorFramework;
 
+import javax.annotation.concurrent.GuardedBy;
+
 /**
  * Factory for BookKeeperLogs.
  */
 @Slf4j
 public class BookKeeperLogFactory implements DurableDataLogFactory {
     //region Members
 
+    // Period of inspection to meet the maximum number of log creation attempts for a given container.
+    private static final Duration LOG_CREATION_INSPECTION_PERIOD = Duration.ofSeconds(60);
+    // Maximum number of log creation attempts for a given container before considering resetting the BK client.
+    private static final int MAX_CREATE_ATTEMPTS_PER_LOG = 2;
+
     private final String namespace;
     private final CuratorFramework zkClient;
     private final AtomicReference<BookKeeper> bookKeeper;
     private final BookKeeperConfig config;
     private final ScheduledExecutorService executor;
+    @GuardedBy("this")
+    @Getter(AccessLevel.PACKAGE)
```

Comment:
How can you have `GuardedBy` and a public getter at the same time? You cannot guarantee thread safety like this. You need to provide APIs that access this while holding locks.

Reply:
Ok, moved the `Getter` to a explicit method tagged with `@VisibleForTesting`.